### PR TITLE
[react/states_and_effects] React revamp - Project Memory Card

### PR DESCRIPTION
**Pull request by Lofty-Brambles**
## Because

This adds a lesson for the react section, updating the old one, according to necessary specs.

## This PR

N/A

## Issue

Closes https://github.com/TheOdinProject/top-meta/issues/266

## Additional Information

N/A

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [ ] ~~The `Because` section summarizes the reason for this PR~~
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: react/states_and_effects/project_memory_card.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+### How The Game Works
+
+Go play around [with this student's solution](https://heldersrvio.github.io/memory-card-game/) to find out for yourself how the Memory game works. Although this example uses cartoon characters, you can use anything you like for your game.
+
+### Assignment
+
+<div class="lesson-content__panel" markdown="1">
+
+1.  Create a new React project with `vite`.
```
Comment (01zulfi) [inline react/states_and_effects/project_memory_card.md:13]: I'm not sure if we need to add this assignment item. "Create a new React project with XYZ tool" isn't future proof.

**Review by 01zulfi — CHANGES_REQUESTED** [review]
The PR title made me do a double take, fixed that.

Just a couple of tiny comments below ⬇️

**Comment by Lofty-Brambles** [comment]
@01zulfi - 👍 I was terribly busy

**Review by 01zulfi — APPROVED** [review]
Looks good 🚀

**Pull request merged**